src/Uno.UWP/Storage/Streams/DataReader.cs

```diff
@@ -0,0 +1,353 @@
+
+namespace Windows.Storage.Streams
+{
+	public partial class DataReader : IDataReader, IDisposable
+	{
+		private IBuffer _buffer = null;
+		private uint _position = 0;
+
+		public ByteOrder ByteOrder { get; set; }
+
+		public uint UnconsumedBufferLength
+		{
+			get => _buffer.Length - _position;
+		}
+
+		private void CheckPosition(int size)
+		{
+			if (_buffer.Length >= (_position + size))
+			{
+				return;
+			}
+
+			throw new IndexOutOfRangeException("Windows.Storage.Streams.DataReader - reading past EOF");
+		}
+
+		public byte ReadByte()
+		{
+			CheckPosition(1);
+			byte value = _buffer.Data[_position];
+			_position++;
+			return value;
+		}
+
+		public IBuffer ReadBuffer(uint length)
+		{
+			CheckPosition((int)length);
+
+			var buffer = new InMemoryBuffer((int)length);
+			for (int i = 0; i < length; i++)
+			{
+				buffer.Data[i] = _buffer.Data[_position];
+				_position++;
+			}
+			return buffer;
+		}
+		public bool ReadBoolean()
+		{
+			byte value = ReadByte();
+			if (value == 0)
+			{
+				return false;
+			}
+			return true;        // although DataWriter uses 0/1, we can also interpret all non-zero values as true;
+		}
+
+		private byte[] ReadBytes(int size)
+		{
+			CheckPosition(size);
+
+			byte[] buffer = new byte[size];
+			for (int i = 0; i < size; i++)
+			{
+				buffer[i] = _buffer.Data[_position];
+				_position++;
+			}
+
+			// maybe we should reverse array (if platform endianess is different than requested)
+			if ((ByteOrder == ByteOrder.LittleEndian && !BitConverter.IsLittleEndian)
+				|| (ByteOrder == ByteOrder.BigEndian && BitConverter.IsLittleEndian))
+			{
+				Array.Reverse(buffer);
+			}
+
+			return buffer;
+		}
+
+		public Guid ReadGuid()
+		{
+			Int32 u32 = ReadInt32();
+			Int16 u16a = ReadInt16();
+			Int16 u16b = ReadInt16();
+			byte[] u64 = ReadBytes(8);
+
+			var value = new Guid(u32, u16a, u16b, u64);
+			return value;
+		}
+		public short ReadInt16() => BitConverter.ToInt16(ReadBytes(2), 0);
+		public int ReadInt32() => BitConverter.ToInt32(ReadBytes(4), 0);
+		public long ReadInt64() => BitConverter.ToInt64(ReadBytes(8), 0);
+		public ushort ReadUInt16() => BitConverter.ToUInt16(ReadBytes(2), 0);
+		public uint ReadUInt32() => BitConverter.ToUInt32(ReadBytes(4), 0);
+		public ulong ReadUInt64() => BitConverter.ToUInt64(ReadBytes(8), 0);
+		public float ReadSingle()
+		{// yes, it is endianness dependant
+			return BitConverter.ToSingle(ReadBytes(4), 0);
+		}
+		public double ReadDouble()
+		{// yes, it is endianness dependant
+			return BitConverter.ToDouble(ReadBytes(8), 0);
+		}
+		public DateTimeOffset ReadDateTime()
+		{
+			long ticks = ReadInt64();
+			var date = new DateTime(1601, 1, 1, 0, 0, 0).ToLocalTime();
+			date = date.AddTicks(ticks);
+
+			return date;
+		}
+		public TimeSpan ReadTimeSpan() => TimeSpan.FromTicks(ReadInt64());
+		public IBuffer DetachBuffer() => _buffer;
+		public void Dispose()
+		{
+			// for DataReader(IBuffer), nothing to do;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305560)